pkg/api/types.go

```diff
@@ -788,6 +788,13 @@ type NodeSpec struct {
 	ExternalID string `json:"externalID,omitempty"`
 }
 
+type NodeIdentity struct {
```

Comment:
Please don't mind the missing comments/descriptions (here and below) it's still a work in progress.